src/main/java/com/google/devtools/build/lib/rules/java/JavaOptions.java

```diff
@@ -597,11 +610,22 @@ public ImportDepsCheckingLevelConverter() {
               + "--java_header_compilation is enabled.")
   public boolean requireJavaToolchainHeaderCompilerDirect;
 
+  private Label getHostJavaBase() {
+    if (hostJavaBase == null) {
+      if (useRemoteJdkAsHostJavaBase) {
```

Comment:
Suggestion: how about implementing this as a `config_setting()`? Then you need less logic in core (only the incompatible flag)

Reply:
This is just temporary. When we switch the flag on, we can remove all the code here.